### ISSUE TITLE
Support for 0.33

### ISSUE THREAD

**Issue by tomtau** [informalsystems/tendermint-rs#184]
Primarily `Block#Header` changed

**Comment by Liamsi** [comment]
ref #88 #181 #182

**Comment by Shivani912** [comment]
https://github.com/informalsystems/tendermint-rs/pull/181#issuecomment-601350728

**Comment by greg-szabo** [comment]
In short, Shivani is working on this on branch [v0.33](https://github.com/informalsystems/tendermint-rs/tree/tendermint/v0.33). We'll merge that branch to master when we're ready with upgrading to 0.33

**Comment by greg-szabo** [comment]
Note: we're planning to use this issue to gather all the outstanding issues that are related to upgrading to v0.33. The issues are usually related to Tendermint Go compatibility over JSONRPC.

Today, I've started going through all RPC struct types and noticed that we sometimes failed to follow the Tendermint-Go implementation's `omitempty` tags. I'm planning to fix that by using the serde annotation `#[serde(default)` (thanks @Liamsi for mentioning that in another issue), instead of the currently implemented `Option<>` way.
This creates cleaner looking code and no `unwrap()` calls.

I'll add other issues into the issue description so we track their progress.

**Comment by greg-szabo** [comment]
Related to 0.33 upgrade:

- Primarily Block#Header changed (mentioned in this issue by @tomtau )
- #88 - rpc incompatibilities - JSON ID incompatibility remains
- #192 - AbciInfo compatibility
- #120 - Run a tendermint node in CI (that runs integration tests with success)
- #4 - RPC client incompatibility with Tendermin 0.32 (most probably this is obsolete already, but it will be with 0.33)

PRs:
- #71 Fix AbciQuery response
- #98 Fix ser/de not symmetric

Branch:
v0.33 branch

**Comment by greg-szabo** [comment]
All issues were resolved in the [v0.33 branch](https://github.com/informalsystems/tendermint-rs/tree/tendermint/v0.33). I just need to figure out what to do with the two PRs from @yihuang .

It seems that I've implemented PR #71 in parallel with some differences. I couldn't actually get a query with `"code": 1` but all other fields are tested in the integration tests, so if this branch is accepted, #71 can be closed without merge.

As for PR #98, I didn't touch that yet. From a high level, it looks good and I think it's mergeable even with the changes in this branch. I'll check it out.
